modules/oe_content_featured_media_field/src/Plugin/Field/FieldWidget/FeaturedMediaEntityBrowserWidget.php

```diff
@@ -0,0 +1,385 @@
+    $details_id = Html::getUniqueId('edit-' . $this->fieldDefinition->getName());
+
+    $element += [
+      '#id' => $details_id,
+      '#type' => 'details',
+      '#open' => !empty($entities) || $this->getSetting('open'),
+      'target_id' => [
+        '#type' => 'hidden',
+        '#id' => $hidden_id,
+        '#attributes' => ['id' => $hidden_id],
+        '#default_value' => implode(' ', array_map(
+          function (EntityInterface $item) {
+            return $item->getEntityTypeId() . ':' . $item->id();
+          },
+          $entities
+        )),
+        '#ajax' => [
+          'callback' => [get_class($this), 'updateWidgetCallback'],
+          'wrapper' => $details_id,
+          'event' => 'entity_browser_value_updated',
+        ],
+      ],
+    ];
+
+    $selection_mode = $this->getSetting('selection_mode');
+
+    // Enable entity browser if requirements for that are fulfilled.
+    if (EntityBrowserElement::isEntityBrowserAvailable($selection_mode, 1, count($entities))) {
+      $persistent_data = $this->getPersistentData();
+
+      $element['entity_browser'] = [
+        '#type' => 'entity_browser',
+        '#entity_browser' => $this->getSetting('entity_browser'),
+        '#cardinality' => 1,
+        '#selection_mode' => $selection_mode,
+        '#default_value' => $entities,
+        '#entity_browser_validators' => $persistent_data['validators'],
+        '#widget_context' => $persistent_data['widget_context'],
+        '#custom_hidden_id' => $hidden_id,
+        '#process' => [
+          ['\Drupal\entity_browser\Element\EntityBrowserElement', 'processEntityBrowser'],
+          [get_called_class(), 'processEntityBrowser'],
+        ],
+      ];
+    }
+
+    $element['#attached']['library'][] = 'entity_browser/entity_reference';
+
+    $field_parents = $element['#field_parents'];
+
+    $element['current'] = $this->displayCurrentSelection($details_id, $field_parents, $entities);
+    // Append the delta to the #name of the remove button because if the same
+    // media entity is referenced, there will be no difference between
+    // buttons.
+    if ($element['current']['items']) {
+      $element['current']['items'][0]['remove_button']['#name'] .= $delta;
+    }
+
+    $element['caption'] = [
+      '#title' => $this->t('Caption'),
+      '#description' => $this->t('The caption that goes with the referenced media.'),
+      '#type' => 'textfield',
+      '#default_value' => $items->offsetExists($delta) ? $items->get($delta)->caption : '',
+      '#element_validate' => [[$this, 'validateCaptionField']],
+    ];
+
+    return $element;
+  }
+
+  /**
+   * Caption field validator.
+   *
+   * @param array $element
+   *   The element.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   * @param array $complete_form
+   *   The entire form.
+   */
+  public function validateCaptionField(array &$element, FormStateInterface $form_state, array &$complete_form): void {
```

Comment:
$complete_form is not used in the function.

Reply:
so? it's passed from the validator.